Code/GraphMol/SubstructLibrary/Wrap/rough_test.py

```diff
@@ -309,6 +309,24 @@ def testBinaryCache(self):
       res = slib.GetMatches(core)
       self.assertEqual(len(res),
                        len([x for x in mols if x.HasSubstructMatch(core, useChirality=True)]))
-        
+
+  def testRingSmartsWithTrustedSmiles(self):
```

Comment:
I think you need more tests that just this one. Please include tests for `r` as well on the atoms, `@` on bonds, and at least one test to make sure your checks also work with recursive SMARTS.

Reply:
Agreed, we should also have C++ tests as well.

Reply:
There is a ring bond check, btw:  "C@C"